crates/ide_assists/src/utils/gen_trait_body.rs

```diff
@@ -0,0 +1,147 @@
+//! This module contains functions to generate default trait impl function bodies where possible.
+
+use syntax::ast::{self, edit::AstNodeEdit, make, AstNode, NameOwner};
+use syntax::ted;
+
+/// Generate custom trait bodies where possible.
+///
+/// Returns `Option` so that we can use `?` rather than `if let Some`. Returning
+/// `None` means that generating a custom trait body failed, and the body will remain
+/// as `todo!` instead.
+pub(crate) fn gen_trait_body(func: &ast::Fn, trait_path: &ast::Path, adt: &ast::Adt) -> Option<()> {
```

Comment:
The function name sounds a bit misleading to me, this doesn't generate a trait body, but the body of a trait function right?